Index: netdata-1.24.0/system/netdata.conf
===================================================================
--- netdata-1.24.0/system/netdata.conf
+++ netdata-1.24.0/system/netdata.conf	2020-09-11 15:33:45.200972503 +0200
@@ -1,14 +1,12 @@
-# netdata configuration
+# This is a minimal configuration file for XCP-ng.
+# You can download a version of this file with ALL possible values, using:
 #
-# You can download the latest version of this file, using:
-#
-#  wget -O /etc/netdata/netdata.conf http://localhost:19999/netdata.conf
-# or
-#  curl -o /etc/netdata/netdata.conf http://localhost:19999/netdata.conf
-#
-# You can uncomment and change any of the options below.
-# The value shown in the commented settings, is the default value.
+#  wget http://localhost:19999/netdata.conf
 #
+# Future updates may overwrite the changes made to this file.
+# When this happens, your previous configuration will be saved to /etc/netdata/netdata.conf.ui.save
+# If you edit this file, make sure to watch for this file after
+# any update that includes the netdata-ui RPM.
 
 [global]
     run as user = netdata
@@ -20,9 +18,17 @@
     process scheduling policy = idle
     OOM score = 1000
 
+    # Don't write anything to disk
+    # netdata 1.19.0 has shown that the 256M limit for dbengine was sometimes broken.
+    # It is supposedly fixed in later versions, but we can't risk filling the whole disk with data.
+    # So on XCP-ng, we default to RAM only, which should cover approximately 1h of stats, using 32M of RAM.
+    memory mode = ram
+
 [web]
     web files owner = root
     web files group = netdata
 
-    # by default do not expose the netdata port
-    bind to = localhost
+    default port = 19999
+    bind to = *
+    allow netdata.conf from = localhost
+
