From 6b0e6a04d78387620be4c0689ef2e8c9ad60c02f Mon Sep 17 00:00:00 2001
From: Thierry Escande <thierry.escande@vates.tech>
Date: Mon, 3 Feb 2025 17:39:32 +0100
Subject: [PATCH] Use local json-c sources
Content-Type: text/plain; charset = "utf-8"
Content-Transfert-Encoding: 8bit

Signed-off-by: Thierry Escande <thierry.escande@vates.tech>
---
 packaging/cmake/Modules/NetdataJSONC.cmake | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/packaging/cmake/Modules/NetdataJSONC.cmake b/packaging/cmake/Modules/NetdataJSONC.cmake
index db18c14..365769a 100644
--- a/packaging/cmake/Modules/NetdataJSONC.cmake
+++ b/packaging/cmake/Modules/NetdataJSONC.cmake
@@ -42,15 +42,13 @@ function(netdata_bundle_jsonc)
 
         if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.28)
                 FetchContent_Declare(json-c
-                        GIT_REPOSITORY ${repo}
-                        GIT_TAG ${tag}
+                        SOURCE_DIR ${JSONC_SOURCES_PATH}
                         CMAKE_ARGS ${NETDATA_CMAKE_PROPAGATE_TOOLCHAIN_ARGS}
                         EXCLUDE_FROM_ALL
                 )
         else()
                 FetchContent_Declare(json-c
-                        GIT_REPOSITORY ${repo}
-                        GIT_TAG ${tag}
+                        SOURCE_DIR ${JSONC_SOURCES_PATH}
                         CMAKE_ARGS ${NETDATA_CMAKE_PROPAGATE_TOOLCHAIN_ARGS}
                 )
         endif()
-- 
2.47.2

