Index: netdata-1.24.0/system/netdata.service.in
===================================================================
--- netdata-1.24.0/system/netdata.service.in
+++ netdata-1.24.0/system/netdata.service.in	2020-09-11 15:40:01.676664376 +0200
@@ -17,6 +17,8 @@
 ExecStartPre=/bin/chown -R netdata:netdata @localstatedir_POST@/cache/netdata
 ExecStartPre=/bin/mkdir -p @localstatedir_POST@/run/netdata
 ExecStartPre=/bin/chown -R netdata:netdata @localstatedir_POST@/run/netdata
+# XCP-ng: call script that will load /etc/sysconfig/iptables_netdata if it exists
+ExecStartPre=/usr/libexec/netdata/xcpng-iptables-restore.sh
 ExecStopPost=@pluginsdir_POST@/reset_netdata_trace.sh
 PermissionsStartOnly=true
 
