Index: netdata-1.18.1/system/netdata.conf
===================================================================
--- netdata-1.18.1/system/netdata.conf
+++ netdata-1.18.1/system/netdata.conf	2019-10-31 16:55:45.842620639 +0100
@@ -1,14 +1,9 @@
-# netdata configuration
+# This is a minimal configuration file for XCP-ng.
+# You can download a version of this file with ALL possible values, using:
 #
-# You can download the latest version of this file, using:
-#
-#  wget -O /etc/netdata/netdata.conf http://localhost:19999/netdata.conf
-# or
-#  curl -o /etc/netdata/netdata.conf http://localhost:19999/netdata.conf
-#
-# You can uncomment and change any of the options below.
-# The value shown in the commented settings, is the default value.
+#  wget http://localhost:19999/netdata.conf
 #
+# Future updates may overwrite the changes made to this file
 
 [global]
     run as user = netdata
@@ -17,8 +12,14 @@
     history = 3600
 
     # by default do not expose the netdata port
+    # avoid changing this value: see the [web] section instead
     bind to = localhost
 
 [web]
     web files owner = root
-    web files group = netdata
+    web files group = root
+
+    default port = 19999
+    bind to = 0.0.0.0 ::
+    allow netdata.conf from = localhost
+
